src/shogun/base/SGObject.h

```diff
@@ -1065,15 +1087,70 @@ class SGObject: public std::enable_shared_from_this<SGObject>
 	 * @param _tag name information of parameter
 	 * @param value new value of parameter
 	 */
-	void update_parameter(const BaseTag& _tag, const Any& value, bool do_checks = true);
+	template <typename T>
+	void update_parameter(const BaseTag& _tag, const T& value, bool do_checks = true)
+	{
+		auto& param = get_parameter(_tag);
+		auto& pprop = param.get_properties();
+		auto& parameter_value = param.get_value();
+		if (pprop.has_property(ParameterProperties::READONLY))
+		        require(!do_checks,
+				"{}::{} is marked as read-only and cannot be modified!",
+		          	get_name(), _tag.name().c_str());
+
+		if (pprop.has_property(ParameterProperties::CONSTRAIN))
+		{
+			auto msg = param.get_constrain_function()(make_any(value));
+			if (!msg.empty())
+			{
+				require(!do_checks,
+					"{}::{} cannot be updated because it must be: {}!",
+					get_name(), _tag.name().c_str(), msg.c_str());
+			}
+		}
+		if constexpr (std::is_same_v<T, Any>)
+		{
+			param.set_value(value);
```

Comment:
this is needed right now for auto init, but I will refactor it in another PR and can then remove this